explainaboard/analysis/performance.py

```diff
@@ -30,35 +33,55 @@ class BucketPerformance:
     bucket_interval: tuple[float, float] | None = None
     bucket_name: str | None = None
 
-    @classmethod
-    def dict_conv(cls, k: str, v: Any) -> Any:
-        """A deserialization utility function.
-
-        It takes in a key corresponding to a
-        parameter name, and dictionary corresponding to a serialized version of that
-        parameter's value, then return the deserialized version of the value.
-
-        Args:
-            k: the parameter name
-            v: the parameter's value
+    def __post_init__(self) -> None:
+        """Validates member values."""
+        has_interval = self.bucket_interval is not None
+        has_name = self.bucket_name is not None
+        if has_interval == has_name:
+            raise ValueError(
+                "Only either `bucket_interavl` or `bucket_name` must has a value. "
+                f"{self.bucket_interval=}, {self.bucket_name=}"
```

Comment:
Just a comment, this is a cool feature that I didn't know of! https://stackoverflow.com/questions/59661904/what-does-equal-do-in-f-strings-inside-the-expression-curly-brackets